pdata/internal/generated_instrumentationscope.go

```diff
@@ -0,0 +1,134 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by "pdata/internal/cmd/pdatagen/main.go". DO NOT EDIT.
+// To regenerate this file run "make genpdata".
+
+package internal
+
+import (
+	otlpcommon "go.opentelemetry.io/collector/pdata/internal/data/protogen/common/v1"
+)
+
+// InstrumentationScope is a message representing the instrumentation scope information.
+//
+// This is a reference type, if passed by value and callee modifies it the
+// caller will see the modification.
+//
+// Must use NewInstrumentationScope function to create new instances.
+// Important: zero-initialized instance is not valid for use.
+type InstrumentationScope struct {
+	commonInstrumentationScope
+}
+
+type MutableInstrumentationScope struct {
+	commonInstrumentationScope
+	preventConversion struct{} // nolint:unused
+}
+
+type commonInstrumentationScope struct {
+	orig *otlpcommon.InstrumentationScope
+}
+
+func NewImmutableInstrumentationScope(orig *otlpcommon.InstrumentationScope) InstrumentationScope {
+	return InstrumentationScope{commonInstrumentationScope{orig}}
+}
+
+func NewMutableInstrumentationScope(orig *otlpcommon.InstrumentationScope) MutableInstrumentationScope {
+	return MutableInstrumentationScope{commonInstrumentationScope: commonInstrumentationScope{orig}}
+}
+
+// NewInstrumentationScope creates a new empty InstrumentationScope.
+//
+// This must be used only in testing code. Users should use "AppendEmpty" when part of a Slice,
+// OR directly access the member if this is embedded in another struct.
+func NewInstrumentationScope() MutableInstrumentationScope {
```

Comment:
We should stick with our guideline/rule `NewMutableInstrumentationScope` since we return `MutableInstrumentationScope`

Reply:
We probably should, yes. Updated